doc/releasing.md

```diff
@@ -0,0 +1,62 @@
+# Release process
+
+Target audience: crate maintainers who wish to release `embedded-graphics-simulator`.
+
+> Please take a cautious approach to this. If any step doesn't feel right or doesn't succeed smoothly, stop and rectify any issues before continuing.
+
+## On GitHub
+
+- Check that all desired PRs are merged and all desired issues are closed/resolved.
+- Check that the latest master build passed in CircleCI.
+
+## On your local machine
+
+- `cd` to the repository root
+- Check that `cargo-release` is installed and available in `$PATH`:
+
+  ```bash
+  cargo release --version
+  ```
+
+- Ensure you have the latest changes with `git switch master` and `git pull --rebase`
+- Check that your local repository is clean with no uncommitted changes and no unpushed commits. Ideally, use `git reset --hard origin/master` to ensure your local state is up to date with `origin/master`. You may need to change `origin` to the name of the remote pointing to <https://github.com/embedded-graphics/simulator>.
+- Before a **stable** release:
+  - Search the repository for any `TODO` or `FIXME` comments. If any need resolving before release, stop this process and fix them with one or more PRs.
+- Check that the crate version in `Cargo.toml` matches the latest released versions on <https://crates.io/crates/embedded-graphics-simulator>.
+- Run `just build` to ensure the build passes locally.
+  - If the build fails for any reason, stop the release process and fix any issues by creating PRs. The upstream master branch must remain the source of truth. Restart this checklist once `just build` passes.
+- Double check the release level (major, minor, patch)
+- Release the crate:
+
+  ```bash
+  cargo release <level>
+  ```
+
+  Where `<level>` is `major`, `minor`, `patch`, or a specific SemVer version number.
```

Comment:
```suggestion
  cargo release --push-remote <push-remote> <level>
  ```

  Where `<level>` is `major`, `minor`, `patch`, or a specific SemVer version number. And where `<push-remote>` is the git remote for the upstream repository `embedded-graphics/simulator`.
```